datamodel/src/main/kotlin/com/hartwig/actin/datamodel/molecular/PanelRecord.kt

```diff
@@ -16,7 +16,10 @@ data class PanelRecord(
     override val hasSufficientQuality: Boolean
 ) : MolecularTest {
 
-    override fun testsGene(gene: String): Boolean {
-        return testedGenes.contains(gene)
+    override fun testsGene(gene: String, molecularTestTargets: List<MolecularTestTarget>): Boolean {
+        if (molecularTestTargets.isEmpty()) {
+            return geneSpecifications[gene] != null
+        }
+        return geneSpecifications[gene]?.containsAll(molecularTestTargets) ?: false
```

Comment:
I don't think the `isEmpty()` check is required (unless you're changing the requirements as Korneel suggested). `containsAll` should return true if `molecularTestTargets` is empty:
```suggestion
        return geneSpecifications[gene]?.containsAll(molecularTestTargets) == true
```